src/Core/Domain/Webservice/Command/AddWebserviceKeyCommand.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Webservice\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Webservice\ValueObject\Key;
+
+/**
+ * Adds new webservice key which is used to access PrestaShop's API
+ */
+class AddWebserviceKeyCommand
+{
+    /**
+     * @var Key
```

Comment:
`WebserviceKey` ? Just `Key` is not ambiguous in your opinion ?

Reply:
well, it depends.. :/ 

first, `WebserviceKey` is actual legacy object model which inside has `key` attribute. So i dont want to confuse developers with this as well.

on the other hand, namespace `Webservice/ValueObject/Key` provides a context. So you can have totally different `Key` in other subdomains and i think its fine.

so, ill let you choose. :stuck_out_tongue:

Reply:
> on the other hand, namespace Webservice/ValueObject/Key provides a context. So you can have totally different Key in other subdomains and i think its fine.

This got me convinced 👍